Add Radshid 2G/4G protocol support

0x7a657573 opened the pull request:
This branch was opened last year but I didn’t have time to finish it until now. I finally revisited the code, fixed several wording / formatting issues in the docs and comments, and made sure everything builds cleanly against the current
master.

Radshid is a company that make 2G/4G trackers, I added the radshid protocol to traccar.
you can find Radshid protocol on https://github.com/traccar/traccar/issues/5235.

tananaev commented on .gitmodules:
```diff
@@ -1,5 +1,5 @@
 [submodule "traccar-web"]
 	path = traccar-web
-	url = ../traccar-web.git
+	url = https://github.com/traccar/traccar-web.git
```
This is not resolved. Please don't mark things as resolved. I will mark them after review.

tananaev commented on src/main/java/org/traccar/protocol/RadshidProtocolDecoder.java:
```diff
@@ -0,0 +1,175 @@
+            long latitude = packetData.readInt();   // Latitude
+            long longitude = packetData.readInt();   // Longitude
+            int altitude = packetData.readShort();  //Altitude
+            int bearing = packetData.readShort();   //Bearing
```
This is not resolved either.

0x7a657573 commented on src/main/java/org/traccar/protocol/RadshidProtocolDecoder.java:
```diff
@@ -0,0 +1,175 @@
+import java.util.LinkedList;
+import java.util.List;
+
+public class RadshidProtocolDecoder extends BaseProtocolDecoder {
```
down

0x7a657573 commented:
Hello @tananaev,
I’ve updated the PR based on your feedback:

Removed unrelated changes

Cleaned up unnecessary variables and comments

Added unit tests for the Radshid protocol

Could you please review it again and let me know if anything else is needed before it can be merged?
Thanks for your time and support! 🙏
